rust/core/src/driver_manager.rs

```diff
@@ -609,15 +618,27 @@ fn load_driver_from_registry(
     entrypoint: Option<&[u8]>,
 ) -> Result<DriverInfo> {
     const ADBC_DRIVER_REGISTRY: &str = "SOFTWARE\\ADBC\\Drivers";
-    let drivers_key = root.open(ADBC_DRIVER_REGISTRY)?;
+    let drivers_key = root
+        .open(ADBC_DRIVER_REGISTRY)
+        .and_then(|k| k.open(driver_name.to_str().unwrap_or_default()))
+        .map_err(|e| {
+            Error::with_message_and_status(
+                format!("Failed to open registry key: {e}"),
+                Status::NotFound,
+            )
+        })?;
+
+    let entrypoint_val = match drivers_key.get_string("entrypoint") {
+        Ok(s) => Some(s.into_bytes()),
+        Err(_) => None,
+    };
```

Comment:
You could use https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.ok